### PR TITLE
173 Improvements

### PR DESCRIPTION
**Pull request by Toxici11i**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl:
tweak: better ways of checking for attacks, blinking and some more feedback to the player
tweak: blink code improved
tweak: SCP-173 starts making people blink further away
tweak: 173 can now attack simplemobs
tweak: Statuecode blind spell now cannot blind the user
tweak: 173 can now see further than humans, through walls and has native nightvision (NV SPELL BEING KEPT JUST INCASE)
tweak: 173 can no longer speak
fix: 173 should no longer blind himself
add: 173 now has functioning AI
tweak: 173 now has a much higher crushforce
/:cl:

## About The Pull Request
Lets 173 attack people in full darkness if blink code isnt running because of how it was done.

## Why It's Good For The Game
This will include other miscellaneous changes that make 173 code more robust, less buggy and more reliable for the player.

### CODE REVIEW

**Review by ghost — DISMISSED** [review]

**Comment by Toxici11i** [comment]
travis failing on soundlen stuff from another PR that was merged

**Comment by Toxici11i** [comment]
temporarily fucked it while im redoing it

**Comment by Toxici11i** [comment]
Travis will fail because yoyos grapeblast PR is causing issues

**Comment by Toxici11i** [comment]
Travis fucked and its not the PR

**Comment by Toxici11i** [comment]
requesting review

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 1]
```diff
@@ -89,3 +98,66 @@
...
+			if(A == src)
+				to_chat(src, "<span class='warning'><I>Why would we waste our energy attacking Ourselves?</I></span>")
+				return
+			if(ishuman(H))
+				visible_message("<span class='danger'>[src] snaps [H]'s neck!</span>")
+				playsound(loc, pick('hippiestation/sound/scpsounds/scp/spook/NeckSnap1.ogg', 'hippiestation/sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
+				return ..()
+			if(!ishuman(H))
+				visible_message("<span class='danger'>[src] Crushes [H] with raw force!</span>")
```
Comment (ghost) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: fix capitalization

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 2]
```diff
@@ -89,3 +98,66 @@
...
+/mob/living/simple_animal/hostile/scp_173/AttackingTarget(var/atom/A) // doesnt work?
+	if(can_be_seen(get_turf(loc)))
+		if(client)
+			to_chat(src, "<span class='warning'>You cannot attack, there are eyes on you!</span>")
+		return FALSE
+	else
+		var/mob/living/H = A
+		if(isliving(A))
+			if(A == src)
+				to_chat(src, "<span class='warning'><I>Why would we waste our energy attacking Ourselves?</I></span>")
```
Comment (ghost) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: fix capitalization

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 3]
```diff
@@ -89,3 +98,66 @@
...
+				return ..()
+			else
+				return FALSE // this is just for error catching
+		else
+			to_chat(src, "<span class='warning'><I>Why would we waste our energy attacking [H]</I></span>")
+			return //doesnt allow 173 to attack apcs
+
+
+/mob/living/simple_animal/hostile/scp_173/DestroyPathToTarget()
+
```
Comment (ghost) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: unneeded space

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 4]
```diff
@@ -89,3 +98,66 @@
...
+
+/mob/living/simple_animal/hostile/scp_173/DestroyPathToTarget()
+	
+	if(!can_be_seen(get_turf(loc)))
+		..()
+
+/mob/living/simple_animal/hostile/scp_173/UnarmedAttack(var/atom/A)
+	if(can_be_seen(get_turf(loc)))
+		if(client)
```
Comment (ghost) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm:141]: why not `if(client && can_be_seen(get_turf(loc)))`?

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 5]
```diff
@@ -89,3 +98,66 @@
...
+		if(isliving(A))
+			if(A == src)
+				to_chat(src, "<span class='warning'><I>Why would we waste our energy attacking Ourselves?</I></span>")
+				return
+			if(ishuman(A))
+				visible_message("<span class='danger'>[src] snaps [H]'s neck!</span>")
+				playsound(loc, pick('hippiestation/sound/scpsounds/scp/spook/NeckSnap1.ogg', 'hippiestation/sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
+			if(!ishuman(A))
+				visible_message("<span class='danger'>[src] Crushes [H] with raw force!</span>")
```
Comment (ghost) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: fix capitalization

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 6]
```diff
@@ -89,3 +98,66 @@
...
+/mob/living/simple_animal/hostile/scp_173/UnarmedAttack(var/atom/A)
+	if(can_be_seen(get_turf(loc)))
+		if(client)
+			to_chat(src, "<span class='warning'>You cannot attack, there are eyes on you!</span>")
+		return FALSE
+	else
+		var/mob/living/H = A
+		if(isliving(A))
+			if(A == src)
+				to_chat(src, "<span class='warning'><I>Why would we waste our energy attacking Ourselves?</I></span>")
```
Comment (ghost) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: fix capitalization

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 7]
```diff
@@ -45,38 +60,32 @@
...
-		playsound(loc, pick('hippiestation/sound/scpsounds/scp/spook/NeckSnap1.ogg', 'hippiestation/sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
-		H.death()
-
-/mob/living/scp_173/Life()
+/mob/living/simple_animal/hostile/scp_173/Life()
 	. = ..()
 	if (isobj(loc))
 		return
-	var/list/our_view = view(src, 13)
+	var/list/our_view = view(src, 23)
```
Comment (ghost) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm:88]: Why is "our_view" 23 tiles?!

**Review by ghost — CHANGES_REQUESTED** [review]

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 2 above)
Comment (Manly-Dwarf) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]:
```suggestion
				to_chat(src, "<span class='warning'><I>You can't hit yourself!</I></span>")
```

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 1 above)
Comment (Manly-Dwarf) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]:
```suggestion
				visible_message("<span class='danger'>[src] crushes [H] with raw force!</span>")
```

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 5 above)
Comment (Manly-Dwarf) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]:
```suggestion
				visible_message("<span class='danger'>[src] crushes [H] with raw force!</span>")
```

**Comment by Manly-Dwarf** [comment]
added suggestions so it'll be really easy to fix the quick shit

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 7 above)
Comment (Toxici11i) [reply hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm:88]: "our view" is just a variable that defines how long scp-173 can make people blink for

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 1 above)
Comment (Toxici11i) [reply hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: the ..() and return is needed or the AI fucks itself

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 4 above)
Comment (Toxici11i) [reply hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm:141]: pointless posting to a non-existent client

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 7 above)
Comment (Toxici11i) [reply hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm:88]: As in, it doesnt set 173's view but simulates a 23 long and 23 wide box going around 173 to detect if people are in range

**Comment by Toxici11i** [comment]
Done, the 23 view thing is where 173 has a box thats 23x23 around it that if people enter, they start blinking

**Comment by AnthonyJNgo** [comment]
You need votes and new reviews. You also have merge conflicts that need to be resolved.

**Comment by Toxici11i** [comment]
> You need votes and new reviews. You also have merge conflicts that need to be resolved.

dont mark things as stale when it isnt reee

**Comment by Toxici11i** [comment]
> You need votes and new reviews. You also have merge conflicts that need to be resolved.

also @AnthonyJNgo check the merge conflict, its where i modified the statuecode

**Comment by Toxici11i** [comment]
```
<<<<<<< patch-19
		if(T && T in targets && T != usr)
=======
		if(T && (T in targets))
>>>>>>> master
```

**Comment by AnthonyJNgo** [comment]
One week with no update/commit/comment from the creator of the PR is long enough to be counted as stale. You've replied so I'll remove that.

**Comment by Toxici11i** [comment]
Not sure why its giving me merge conflicts when all ive done is included an extra bit onto statue.dm as a code improvement

**Comment by AnthonyJNgo** [comment]
Find the conflict and see what the issue is. Most likely we have merged a pr that does something to one of the related files which is where the conflict is coming from.

**Comment by Toxici11i** [comment]
> Find the conflict and see what the issue is. Most likely we have merged a pr that does something to one of the related files which is where the conflict is coming from.

nope lmao, thats why im confused

**Comment by Toxici11i** [comment]
ok fixed it seems

**Comment by danhibiki** [comment]
I’m ok with these tweaks +1

**Comment by Toxici11i** [comment]
what the fuck this was merge ready and had 0 conflicts

**Comment by Toxici11i** [comment]
fucking niggerdel

**Pull request closed without merging**